### ISSUE TITLE
issue #242 is still not fixed

### ISSUE THREAD

**Issue by marcelh-gh** [wallabyjs/quokka#247]
1.0.144 still gives all kinds of weird errors on all my projects.
You might have fixed some issue(s) but clearly there are more regarding your new handling of tsconfig

As an example:
```
​​​​Quokka #1 (node: v10.5.0)​​​​
​​Unexpected token } in JSON at position 686​​
```
This happens in a tiny project with almost nothing configured in tsconfig. Yes, the project contains a few json files, but that should not bother Quokka.

Also, the new Quokka does not seem to handle Node.js stuff like "path" properly:
```
​​​​Quokka #1 (node: v10.5.0, TypeScript: v3.0.3)​​​​
​​​​
​​Unexpected token *​​
  at ​​​new Script​​​ ​vm.js:74​
  at ​​​createScript​​​ ​vm.js:246​
  at ​​​Object.runInThisContext​​​ ​vm.js:298​
  at ​​​Module._compile​​​ ​internal/modules/cjs/loader.js:670​

​Install "path" package for the current quokka file​
​Install "path" package into the project​
```

_it would really help if you would have some kind of changelog available, so users could see what areas in particular you have been working on._

**Comment by ArtemGovorov** [comment]
Unfortunately we can't reproduce any of the above mentioned issues.

> This happens in a tiny project with almost nothing configured in tsconfig. 

Here is a tiny sample project with a simple tsconfig.json - https://github.com/wallabyjs/quokka-typescript-import-sample. Could you please clone the repo, `npm i` and run Quokka on a new Quokka file or the `quokka.ts` file in the project?

> Also, the new Quokka does not seem to handle Node.js stuff like "path" properly:

Could you please share the code sample that produces the error?

**Comment by marcelh-gh** [comment]
@ArtemGovorov thanks for the lightning fast reply! :-)
I cloned the repo and it works. Also updated typescript to 3.03 - still works.
I'll look into it some more ant let you know...

**Comment by marcelh-gh** [comment]
ok. just found out that the error about a json file was actually about the tsconfig. there was a trailing comma which caused that error. 
_suggestion: it would help to display the filename if you encounter an issue_

here is the tsconfig of the tiny project.
```
{
    "compilerOptions": {
        "module": "commonjs",
        "strict": true,
        "strictPropertyInitialization": false,
        "noUnusedParameters": true,
        "removeComments": false,
        "preserveConstEnums": true,
        "sourceMap": true,
        "noEmit": true,
        "target": "ES2018",
        "lib": [
            "es2018"
        ],
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "resolveJsonModule": true,
        "esModuleInterop": true,
        "pretty": true,
        "baseUrl": ".",
        "paths": {
            "@SHARED/*": [
                "../shared/*"
            ]
        }
    },
    "exclude": [
        "node_modules",
        "compiledJS",
        "documentation",
        "unitTesting",
        "lib",
        "bundle.js"
    ]
}
```
With the trailing comma fixed, Quokka now reports no error but also does not do anything even for a file like
`const A = 5;
const B = A * 2;`

Do you see any issues with this tsconfig?

**Comment by ArtemGovorov** [comment]
Thanks for sharing the config. I can see the issue, it is:

```
"noEmit": true,
```

with this setting TypeScript compiler doesn't emit any JavaScript, so Quokka has nothing to run.

What we should probably do is ignore the setting in Quokka (however I suspect you may still have some issues with other tools, like `ts-node`).

**Comment by marcelh-gh** [comment]
Ignoring the no emit would make sense. I am using ts-node and it works just fine.
One of the reasons I use `noEmit` here is that I don't need the js, everything get's executed via ts-node...

**Comment by ArtemGovorov** [comment]
Thanks for your feedback. 

> Ignoring the no emit would make sense.

Fair enough, we have just released a new version (v1.0.145) where `noEmit` is ignored.

Will also look into the trailing comma case later.

**Comment by marcelh-gh** [comment]
@ArtemGovorov Thanks a lot! :-)
